enterprise/server/remote_execution/executor/executor.go

```diff
@@ -341,7 +341,7 @@ func (s *Executor) ExecuteTaskAndStreamResults(ctx context.Context, st *repb.Sch
 		log.CtxErrorf(ctx, "Failed to publish ExecuteResponse: %s", err)
 		return finishWithErrFn(err)
 	}
-	if cmdResult.Error == nil {
+	if cmdResult.Error == nil && cmdResult.ExitCode == 0 {
```

Comment:
This will cause a persistent worker process to be killed if it tries to compile a source file with an error, which is perfectly OK and should not kill the persistent worker.

It's also not clear whether this is the right approach for workflow runs and remote bazel runs, because let's say we spend a lot of time building the full action graph but then there's a small compile error - it means we'll throw away a bazel cache which is mostly good, and so the next compile that just fixes the small build error will have to start from scratch.

Reply:
Ack. I wonder if we should filter on certain error codes. It kind of seems like if there's a build error we might want to recycle it, but if there's a setup error it's probably not worth it (i.e. anything related to git)

Re your other comment: Yes we can avoid applying bad patches, but there are other cases - i.e. If there's a merge conflict between two branches, if there's a network error and we can't fetch a branch etc.